packages/vscode-extension/lib/metro_helpers.js

```diff
@@ -11,7 +13,16 @@ function requireFromAppDir(module, options) {
 
 function resolveFromAppDir(module, options) {
   const paths = options && options.paths ? [...options.paths, appRoot] : [appRoot];
-  return require.resolve(module, { paths });
+  try {
+    return require.resolve(module, { paths });
+  } catch (e) {
+    // if there was an error, we try to resolve the module via pnpm resolution
+  }
+
+  // we try pnpm resolution here as it is our only "special-cased" resolution mechanism
+  // if we need to support other package managers, we can add more resolution mechanisms here
+  // and switch between them based on some criteria (e.g. presence of lock file)
+  return resolvePnpmPackageFromAppRoot(module);
```

Comment:
Is this really necessary?
Which dependency we try to import is causing problems?